interfaces/builtin/desktop_launch.go

```diff
@@ -0,0 +1,64 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const desktopLaunchSummary = `allows snaps to identify and launch other snaps`
+
+const desktopLaunchBaseDeclarationPlugs = `
+  desktop-launch:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const desktopLaunchBaseDeclarationSlots = `
+  desktop-launch:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const desktopLaunchConnectedPlugAppArmor = `
+# Description: Can identify and launch other snaps.
+
+# Access to the desktop files installed by snaps
+/var/lib/snapd/desktop/applications/{,*} r,
+
+#include <abstractions/dbus-session-strict>
+
+dbus (send)
+    bus=session
+    path=/io/snapcraft/Launcher
+    interface=io.snapcraft.Launcher
```

Comment:
I'd appreciate a concrete suggestion for how to resolve this.

Reply:
How about:
```
dbus (send)
    bus=session
    path=/io/snapcraft/Launcher
    interface=io.snapcraft.PrivilegedDesktopLauncher
```

@jhenstridge - thoughts? @AlanGriffiths - if desired, feel free to cut over to my suggestion if we don't hear back from @jhenstridge by the time you want to do it (since changing it to the final interface name should be easy enough).

Reply:
> I've seen this, but not implemented it yet as I think it needs splitting the code out into a bunch of new files. Something I'd like consensus on the name before doing.

I've had a brief try at this, but it didn't join up over dbus. Could someone that understands the wiring needed offer help? (I'm back on Tuesday after the holiday weekend.)